### PR TITLE
Remove multiple src/dst nodetable support

### PR DESCRIPTION
**Pull request by acquamarin**
Solves issue #1221 : 
1. This PR removes the support of multiple src/dst tables in a relation.
(e.g. we no longer support relation like: (person|animal)-[:knows]->(person)).
2. Refactors test files.

### CODE REVIEW

File: src/include/storage/store/rel_table.h [hunk 1]
```diff
@@ -145,14 +125,16 @@ class DirectedRelTableData {
...
+    unordered_map<property_id_t, unique_ptr<Column>> propertyColumns;
+    unique_ptr<AdjColumn> adjColumn;
+    unordered_map<property_id_t, unique_ptr<Lists>> propertyLists;
+    unique_ptr<AdjLists> adjLists;
     table_id_t tableID;
+    table_id_t boundTableID;
     RelDirection direction;
     ListsUpdatesStore* listsUpdatesStore;
     BufferManager& bufferManager;
+    bool isSingleMultiplicityInDirection;
```
Comment (andyfengHKU) [inline src/include/storage/store/rel_table.h:137]: Add a TODO for me to remove this eventually

File: src/storage/store/rel_table.cpp [hunk 2]
```diff
@@ -156,40 +124,38 @@ void DirectedRelTableData::scanLists(Transaction* transaction, RelTableScanState
...
+        auto propertyColumn = getPropertyColumn(i);
         propertyColumn->writeValues(boundVector, relPropertyVectors[i]);
     }
 }
 
 void DirectedRelTableData::deleteRel(const shared_ptr<ValueVector>& boundVector) {
     auto boundNode =
         boundVector->getValue<nodeID_t>(boundVector->state->selVector->selectedPositions[0]);
-    if (!adjColumns.contains(boundNode.tableID)) {
+    if (!isSingleMultiplicity()) {
```
Comment (andyfengHKU) [inline src/storage/store/rel_table.cpp:152]: this is not related to this PR. But should we error in this case instead of return? Ditto for insert and update

File: src/storage/store/rel_table.cpp [hunk 3]
```diff
@@ -245,24 +213,24 @@ void RelTable::initializeData(RelTableSchema* tableSchema) {
     bwdRelTableData->initializeData(tableSchema, wal);
 }
 
-vector<AdjLists*> RelTable::getAdjListsForNodeTable(table_id_t boundNodeTableID) {
+vector<AdjLists*> RelTable::getAllAdjLists(table_id_t boundTableID) {
     vector<AdjLists*> retVal;
-    if (fwdRelTableData->hasAdjLists(boundNodeTableID)) {
-        retVal.push_back(fwdRelTableData->getAdjLists(boundNodeTableID));
+    if (!fwdRelTableData->isSingleMultiplicity() && fwdRelTableData->isBoundTable(boundTableID)) {
```
Comment (andyfengHKU) [inline src/storage/store/rel_table.cpp:218]: Up to u, but I guess we can still have isAdjLists(tableID) and is AdjColumn(tableID) API

File: test/test_files/tinysnb/agg/multi_label.test [hunk 4]
```diff
@@ -1,33 +1,9 @@
...
+7:2|2|1905-12-12
 
 -NAME MultiLabelAggTest2
--QUERY MATCH (a:person)<-[e1:marries|:mixed|studyAt]-(b:person)-[e2:knows]->(c:person) WHERE b.ID = 7 RETURN ID(e1), COUNT(*), MIN(e2.date)
----- 2
-7:3|2|1905-12-12
-8:2|2|1905-12-12
-
--NAME MultiLabelAggTest3
--QUERY MATCH (a:person)-[e1:marries|:mixed|studyAt]->(b:person) RETURN e1.year, COUNT(*)
```
Comment (andyfengHKU) [inline test/test_files/tinysnb/agg/multi_label.test:14]: I think mixed can be rewrite as workAt|meets except there is one property missing. Can u rewrite multi-label test that are deleted due to `mixed`. We are less confidence on multi-labeled feature at this point.

File: src/include/storage/store/rels_statistics.h [hunk 5]
```diff
@@ -71,23 +54,15 @@ class RelsStatistics : public TablesStatistics {
...
-    void setNumRelsPerDirectionBoundTableID(
-        table_id_t tableID, vector<map<table_id_t, atomic<uint64_t>>>& directionNumRelsPerTable);
+    void updateNumRelsByValue(table_id_t relTableID, int64_t value);
 
     offset_t getNextRelOffset(Transaction* transaction, table_id_t tableID);
 
 protected:
     inline string getTableTypeForPrinting() const override { return "RelsStatistics"; }
 
     inline unique_ptr<TableStatistics> constructTableStatistic(TableSchema* tableSchema) override {
```
Comment (ray6080) [inline src/include/storage/store/rels_statistics.h:64]: Do we still need this function? It seems this should be removed now.

File: test/CMakeLists.txt [hunk 6]
```diff
@@ -6,7 +6,7 @@ FetchContent_Declare(
         GIT_TAG release-1.11.0
 )
 FetchContent_MakeAvailable(googletest)
-add_library(GTest::GTest INTERFACE IMPORTED)
+add_library(GTest::GTest INTERFACE IMPORTED runner/e2e_create_rel_test.cpp)
```
Comment (ray6080) [inline test/CMakeLists.txt:9]: Revert the change here.

**Review by ray6080 — APPROVED** [review]
Can you upgrade project version to `0.0.1.4` in CMake file? So you can trigger an auto-reload of the benchmark dataset.

File: src/include/storage/storage_structure/in_mem_page.h [hunk 7]
```diff
@@ -5,13 +5,15 @@
...
 #include "common/null_mask.h"
-#include "storage/node_id_compression_scheme.h"
+#include "common/types/internal_id_t.h"
 
 using namespace kuzu::common;
 
 namespace kuzu {
 namespace storage {
 
+using namespace std;
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply src/include/storage/storage_structure/in_mem_page.h:15]: Removed

File: src/include/storage/store/rel_table.h [hunk 8]
```diff
@@ -9,14 +9,6 @@
 namespace kuzu {
 namespace storage {
 
-// TODO(Guodong): remove the distinction between AdjColumn and Column, also AdjLists and Lists.
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply src/include/storage/store/rel_table.h:12]: reverted this TODO

File: src/include/storage/store/rel_table.h (hunk 1 above)
Comment (acquamarin) [reply src/include/storage/store/rel_table.h:137]: Added a todo

File: src/storage/store/rel_table.cpp (hunk 2 above)
Comment (acquamarin) [reply src/storage/store/rel_table.cpp:152]: No, the operator doesn't know whether the edge is stored as list or column, it needs to check whether we should delete from column

File: test/test_files/tinysnb/agg/multi_label.test (hunk 4 above)
Comment (acquamarin) [reply test/test_files/tinysnb/agg/multi_label.test:14]: If i removed the mixed table, e.g. the MultiLabelAggTest3 has no difference between MultiLabelAggTest2. I rewrote some multi-labeled tests which are worthing keep.

File: src/include/storage/copy_arrow/copy_rel_arrow.h [hunk 9]
```diff
@@ -123,14 +118,13 @@ class CopyRelArrow : public CopyStructuresArrow {
...
-    vector<map<table_id_t, unique_ptr<atomic_uint64_vec_t>>> directionTableListSizes{2};
-    vector<map<table_id_t, atomic<uint64_t>>> directionNumRelsPerTable{2};
-    vector<map<table_id_t, NodeIDCompressionScheme>> directionNodeIDCompressionScheme{2};
-    vector<table_adj_in_mem_columns_map_t> directionTableAdjColumns{2};
-    vector<table_property_in_mem_columns_map_t> directionTablePropertyColumns{2};
-    vector<table_adj_in_mem_lists_map_t> directionTableAdjLists{2};
-    vector<table_property_in_mem_lists_map_t> directionTablePropertyLists{2};
-    unordered_map<uint32_t, unique_ptr<InMemOverflowFile>> overflowFilePerPropertyID;
+    atomic<uint64_t> numRels = 0;
+    vector<unique_ptr<atomic_uint64_vec_t>> directionTableListSizes{2};
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply src/include/storage/copy_arrow/copy_rel_arrow.h:122]: renamed

File: src/include/storage/storage_structure/column.h [hunk 10]
```diff
@@ -215,39 +215,37 @@ class RelIDColumn : public Column {
 class AdjColumn : public Column {
 
 public:
-    AdjColumn(const StorageStructureIDAndFName& structureIDAndFName, BufferManager& bufferManager,
-        const NodeIDCompressionScheme& nodeIDCompressionScheme, WAL* wal)
-        : Column{structureIDAndFName, DataType(INTERNAL_ID),
-              nodeIDCompressionScheme.getNumBytesForNodeIDAfterCompression(), bufferManager, wal},
-          nodeIDCompressionScheme(nodeIDCompressionScheme){};
+    AdjColumn(const StorageStructureIDAndFName& structureIDAndFName, table_id_t commonNbrTableID,
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply src/include/storage/storage_structure/column.h:218]: renamed

File: src/include/storage/storage_structure/column.h [hunk 11]
```diff
@@ -215,39 +215,37 @@ class RelIDColumn : public Column {
...
-        nodeIDCompressionScheme.writeNodeID(
-            walPageInfo.frame + mapElementPosToByteOffset(walPageInfo.posInPage),
-            vectorToWriteFrom->getValue<nodeID_t>(posInVectorToWriteFrom));
+        *(walPageInfo.frame + mapElementPosToByteOffset(walPageInfo.posInPage)) =
+            vectorToWriteFrom->getValue<nodeID_t>(posInVectorToWriteFrom).offset;
     }
 
 private:
-    NodeIDCompressionScheme nodeIDCompressionScheme;
+    table_id_t commonNbrTableID;
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply src/include/storage/storage_structure/column.h:248]: renamed

File: src/include/storage/storage_structure/lists/lists.h [hunk 12]
```diff
@@ -185,13 +184,12 @@ class AdjLists : public Lists {
 
 public:
     AdjLists(const StorageStructureIDAndFName& storageStructureIDAndFName,
-        BufferManager& bufferManager, NodeIDCompressionScheme nodeIDCompressionScheme, WAL* wal,
+        table_id_t commonNbrTableID, BufferManager& bufferManager, WAL* wal,
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply src/include/storage/storage_structure/lists/lists.h:187]: renamed

File: src/include/storage/storage_structure/lists/lists.h [hunk 13]
```diff
@@ -226,7 +220,7 @@ class AdjLists : public Lists {
         ListHandle& listHandle, const shared_ptr<ValueVector>& valueVector);
 
 private:
-    NodeIDCompressionScheme nodeIDCompressionScheme;
+    table_id_t commonNbrTableID;
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply src/include/storage/storage_structure/lists/lists.h:223]: renamed

File: src/include/storage/store/rels_statistics.h (hunk 5 above)
Comment (acquamarin) [reply src/include/storage/store/rels_statistics.h:64]: Although this function calls the relStatistics constructor which takes 0 parameter, this 0 parameter constructor initializes the nextRelOffset field. So we need this function.

File: test/CMakeLists.txt (hunk 6 above)
Comment (acquamarin) [reply test/CMakeLists.txt:9]: reverted
